server/log_operation_manager.go

```diff
@@ -286,13 +326,17 @@ func (l *LogOperationManager) masterFor(ctx context.Context, allIDs []int64) ([]
 	return held, nil
 }
 
-func (l *LogOperationManager) updateHeldIDs(logIDs, allIDs []int64) {
+func (l *LogOperationManager) updateHeldIDs(ctx context.Context, logIDs, allIDs []int64) {
 	l.heldMutex.Lock()
 	defer l.heldMutex.Unlock()
 	if !reflect.DeepEqual(logIDs, l.lastHeld) {
 		l.lastHeld = make([]int64, len(logIDs))
 		copy(l.lastHeld, logIDs)
-		glog.Infof("now acting as master for %d / %d: %s", len(logIDs), len(allIDs), util.HeldInfo(logIDs, allIDs))
+		heldInfo := l.heldInfo(ctx, logIDs)
+		glog.Infof("now acting as master for %d / %d, %s", len(logIDs), len(allIDs), heldInfo)
+		if l.info.Registry.SetProcessStatus != nil {
```

Comment:
Adding to my previous suggestion of moving SetProcessStatus to LogOperationManager, I'd also suggest setting the default as an "empty" function, so we can get rid of the if and always call it.

Reply:
`SetProcessStatus` is still in the registry, but I've assumed it's non-nil so the `if` can be dropped.

Reply:
I've changed my mind, I prefer the original.